domainservice/src/main/kotlin/com/example/delivery/controller/AbstractNatsController.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.delivery.controller
+
+import com.example.delivery.annotaion.NatsController
+import com.example.delivery.annotaion.NatsHandler
+import com.google.protobuf.GeneratedMessage
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import io.nats.client.Dispatcher
+import jakarta.annotation.PostConstruct
+import reactor.core.publisher.Mono
+import java.lang.reflect.ParameterizedType
+
+abstract class AbstractNatsController(
```

Comment:
I like this approach

However, it will be not trivial to rewrite it to use ajax lib as we create 1 class per endpoint